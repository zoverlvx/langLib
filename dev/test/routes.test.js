import React from "react";
import {shallow, mount} from "enzyme";
import {MemoryRouter} from "react-router-dom";
import toJson from "enzyme-to-json";
import Routes from "../js/components/routes";

const component = shallow(<Routes />);
const tree = toJson(component);

describe("<Routes />", () => {
    it("renders 1 <Routes /> component", () => {
	    console.log(tree); 
		expect(component).toHaveLength(1);
	});

	it("provides a snapshot of Routes' tree", () => { 
		expect(tree).toMatchSnapshot();
	});

	it("should render the LanguageBtnCreator when visiting /", () => {
        const component = mount(
            <MemoryRouter initialEntries={["/"]} initialIndex={0}>
                <Routes />
            </MemoryRouter>
        	)
        expect(component.find(Home).length).toBe(1)
	});

	// it("should render ListLanguageAspects on /:language", () => {
    // this is currently the main responsibility of Redux
	// });

    // it("should render LanguageDetail when visiting /:language/:part-of-speech", () => {
    
    // });

});