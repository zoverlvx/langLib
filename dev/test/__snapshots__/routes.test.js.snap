// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> provides a snapshot of Routes' tree 1`] = `
<div>
  <BrowserRouter>
    <Route
      component={[Function]}
      path="/"
    />
    <Route
      component={[Function]}
      path=":language"
    />
    <Route
      component={[Function]}
      path=":language/:part-of-speech"
    />
  </BrowserRouter>
</div>
`;
